src/commons/application/Application.tsx

```diff
@@ -70,6 +73,47 @@ const Application: React.FC = () => {
     };
   }, [isPWA, isMobile]);
 
+  // Effect to fetch the latest user info and course configurations from the backend on refresh,
+  // if the user was previously logged in
+  React.useEffect(() => {
+    // Polyfill confirm() to instead show as VSCode notification
+    window.confirm = () => {
+      console.log('You gotta confirm!');
+      return true;
+    };
+
+    const message = Messages.WebviewStarted();
+    sendToWebview(message);
+
+    window.addEventListener('message', event => {
+      const message: MessageType = event.data;
+      // Only accept messages from the vscode webview
+      if (!event.origin.startsWith('vscode-webview://')) {
+        return;
```

Comment:
I know this is a guard, but shall we log such an event?

Technically we might be able to configure our deployment with `X-Frame-Options: DENY` to prevent embedding but let's look into it another time

Reply:
Note to self: also add Content-Security-Policy frame-ancestors 'none';

Reply:
Adding header `X-Frame-Options: DENY` does not compromise the usability of the VSC extension, but `Content-Security-Policy: frame-ancestors 'none';` will cause the request to be blocked within VSC.